From 80b0e03a8e38618a59ba36a4e96ea572613e6b62 Mon Sep 17 00:00:00 2001
From: Daan De Meyer <daan.j.demeyer@gmail.com>
Date: Mon, 6 Jan 2025 16:28:39 +0100
Subject: [PATCH] fmf: Insist on mkosi sha being available

Let's assume the workflow file containing the mkosi always being
available.

Also fix the detection while we're at it by not leaving the systemd
directory.

(cherry picked from commit 9e76e9c9dc211440e79a7e27600593169597cc41)
---
 test/fmf/integration-tests/test.sh | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/test/fmf/integration-tests/test.sh b/test/fmf/integration-tests/test.sh
index 51cbab6ec9..f16bd84cba 100755
--- a/test/fmf/integration-tests/test.sh
+++ b/test/fmf/integration-tests/test.sh
@@ -17,20 +17,14 @@ if [[ -n "${PACKIT_SOURCE_URL:-}" ]]; then
     git merge "pr/$PACKIT_SOURCE_BRANCH"
 fi
 git log --oneline -5
-popd
 
 # Now prepare mkosi, possibly at the same version required by the systemd repo
 git clone https://github.com/systemd/mkosi
-# If we have it, pin the mkosi version to the same one used by Github Actions, to ensure consistency
-if [ -f .github/workflows/mkosi.yml ]; then
-    mkosi_hash="$(grep systemd/mkosi@ .github/workflows/mkosi.yml | sed "s|.*systemd/mkosi@||g")"
-    git -C mkosi checkout "$mkosi_hash"
-fi
+mkosi_hash="$(grep systemd/mkosi@ .github/workflows/mkosi.yml | sed "s|.*systemd/mkosi@||g")"
+git -C mkosi checkout "$mkosi_hash"
 
 export PATH="$PWD/mkosi/bin:$PATH"
 
-pushd systemd
-
 # shellcheck source=/dev/null
 . /etc/os-release || . /usr/lib/os-release
 
